ttyplot.c

```diff
@@ -616,17 +584,21 @@ int main(int argc, char *argv[]) {
             redraw_needed = true;
 
         // Handle signals.
-        if (sigint_pending) {
-            break;
-        }
-        if (sigwinch_pending) {
-            sigwinch_pending = 0;
-            endwin();
-            initscr();
-            erase();
-            refresh();
-            gethw();
-            redraw_needed = true;
+        if (select_ret > 0 && FD_ISSET(signals[0], &read_fds)) {
+            int signum;
+            int count = read(signals[0], &signum, sizeof(signum));
+            if (count > 0) {
```

Comment:
Is there some guarantee here that read reads full `sizeof(int)` bytes every time? if yes, can we add an `assert`, if not, can we handle the case of `count > 0 && count < sizeof(signum)`?

Reply:
My understanding is that `read` gives either the amount of data requested, or the amount available on the pipe, whichever is smaller. But now that you mention it, I am not too comfortable relying on that.

I changed the code to transmit a single byte, which is always atomic. We are only transmitting small numbers anyway: `SIGINT` is&nbsp;2, `SIGWINCH` is&nbsp;28, both on Linux and BSD. I do not know of any signal number larger than&nbsp;33.

Reply:
> I changed the code to transmit a single byte, which is always atomic.

@edgar-bonet very nice idea! :+1: